src/System.Private.ServiceModel/tests/Stress/SharedPoolsOfWCFObjects/SharedPoolsOfWCFObjects.cs

```diff
@@ -0,0 +1,612 @@
+                    {
+                        s_CallChannelStats.CallActionAndRecordStats(() =>
+                        {
+                            channel.GetData(44);
+                        });
+                    }
+                }
+            }
+        }
+    }
+
+    // One of the most common scenario: create a pool of channel factories, use each factory to create a pool of channels,
+    // and then use all pooled channels for the duration of the stress run
+    public class PooledFactoriesAndChannels
+    {
+        private static CallStats s_CreateChannelStats = new CallStats();
+        private static CallStats s_CallChannelStats = new CallStats();
+        private static CallStats s_CloseChannelStats = new CallStats();
+        private static CallStats s_CloseFactoryStats = new CallStats();
+        private static PoolOfThings<
+            FactoryAndPoolOfItsObjects<
+                ChannelFactory<WcfService1.IService1>,
+                WcfService1.IService1>
+            > s_pooledFactoriesAndChannels;
+
+        static PooledFactoriesAndChannels()
+        {
+            s_pooledFactoriesAndChannels = StaticDisposablesHelper.AddDisposable(
+                new PoolOfThings<FactoryAndPoolOfItsObjects<ChannelFactory<WcfService1.IService1>, WcfService1.IService1>>(
+                    maxSize: 3, // # of pooled FactoryAndPoolOfItsObjects
+                    createInstance: () => new FactoryAndPoolOfItsObjects<ChannelFactory<WcfService1.IService1>, WcfService1.IService1>(
+                        createFactoryInstance: () =>
+                            TestHelpers.CreateChannelFactory<WcfService1.IService1>(TestHelpers.CreateEndPointAddress(), TestHelpers.CreateBinding()),
+                        destroyFactoryInstance: (chf) =>
+                            s_CloseFactoryStats.CallActionAndRecordStats(() => TestHelpers.CloseFactory(chf)),
+                        maxPooledObjects: 3, // # of pooled channels within each pooled FactoryAndPoolOfItsObjects
+                        createObject: (chf) =>
+                            s_CreateChannelStats.CallFuncAndRecordStats(TestHelpers.CreateChannel, chf),
+                        destroyObject: (ch) =>
+                            s_CloseChannelStats.CallActionAndRecordStats(() => TestHelpers.CloseChannel(ch))
+                        ),
+                    destroyInstance: (_fapoic) => _fapoic.Destroy()));
+        }
+
+        public static void UseChannelsInPooledFactoriesAndChannels()
+        {
+            foreach (var factoryAndPoolOfItsChannels in s_pooledFactoriesAndChannels.GetAllPooledInstances())
+            {
+                foreach (var channel in factoryAndPoolOfItsChannels.ObjectsPool.GetAllPooledInstances())
+                {
+                    s_CallChannelStats.CallActionAndRecordStats(() =>
+                    {
+                        channel.GetData(44);
+                    });
```

Comment:
For future consideration -- make '44' some visible const or static readonly field somewhere.